oshmem/mca/spml/ucx/spml_ucx.c

```diff
@@ -691,23 +691,40 @@ int mca_spml_ucx_add_procs(oshmem_group_t* group, size_t nprocs)
         }
     }
 
+    indices = malloc(nprocs * sizeof(int));
+    if (!indices) {
+        goto error;
+    }
+
+    for (i = 0; i < nprocs; i++) {
+        indices[i] = i;
+    }
+
+    srand((unsigned int)time(NULL));
+
     /* Get the EP connection requests for all the processes from modex */
-    for (n = 0; n < nprocs; ++n) {
-        i = (my_rank + n) % nprocs;
+    for (i = nprocs - 1; i >= 0; --i) {
+        /* Fisher-Yates shuffle algorithm */
+        if (i > 0) {
+            j = rand() % (i + 1);
+            temp = indices[i];
+            indices[i] = indices[j];
+            indices[j] = temp;
+        }
 
         ep_params.field_mask = UCP_EP_PARAM_FIELD_REMOTE_ADDRESS;
-        ep_params.address    = (ucp_address_t *)mca_spml_ucx.remote_addrs_tbl[0][i];
+        ep_params.address    = (ucp_address_t *) mca_spml_ucx.remote_addrs_tbl[0][indices[i]];
 
         err = ucp_ep_create(mca_spml_ucx_ctx_default.ucp_worker[0], &ep_params,
-                &mca_spml_ucx_ctx_default.ucp_peers[i].ucp_conn);
+                            &mca_spml_ucx_ctx_default.ucp_peers[indices[i]].ucp_conn);
         if (UCS_OK != err) {
             SPML_UCX_ERROR("ucp_ep_create(proc=%zu/%zu) failed: %s", n, nprocs,
```

Comment:
also update log line index (i and indices[i])

Reply:
ah right, already up-to-date. we could add `indices[proc_index]` to the log along with iteration since it is randomized.

Reply:
done